### PR TITLE
A new output plugin that supports synchronization of object structures to elasticsearch

### PR DESCRIPTION
**Pull request by treeyh**
Description：
- this plugin is still in beta
- it supports Elasticsearch version 7.x and 6.x
- it support for dynamically creating indexes
- it supports one-to-one and one-to-many table relationships.
- temporarily does not support many-to-many relationships

configuration demo：
```toml

[output]
type = "esmodel"

[output.config]
# Whether to ignore 400(bad request) responses.
# Elasticsearch will return 400 when the index name is invalid or the document parsing error.
# it is disabled by default, so the synchronization will be failed when the remote server returned a 400 error.
ignore-bad-request = true

#
# The server configuration of Elasticsearch
# Required
#
[output.config.server]
urls = ["http://192.168.1.152:9200"]
sniff = false
# http timeout, default is 1000ms
timeout = 500

#
# The basic auth configuration
# Optional
#
[output.config.server.auth]
username = ""
password = ""

# master table strategy
[[output.config.routes]]
match-schema = "test"
# main table
match-table = "student"
# index name
index-name="student_index"
# Type name, 7.x item is invalid
type-name="student"
#shards number
shards-num=1
#replicas number
replicas-num=0
# retry count
retry-count=3
#Included columns, default all
include-column = []
#Excluded column, default empty
exclude-column = []

# columns convert strategy
[output.config.routes.convert-column]
name = "studentName"

# one-to-one table strategy
[[output.config.routes.one-one]]
match-schema = "test"
match-table = "student_detail"
# foreign key column
fk-column = "student_id"
#Included columns, default all
include-column = []
#Excluded column, default empty
exclude-column = []
# Mode, 1: sub-object, 2: index extend
mode = 2
# Property object name, valid when mode is 1.
property-name = "studentDetail"
# Attribute prefix, valid when mode is 2.
property-pre = "sd_"

# columns convert strategy
[output.config.routes.one-one.convert-column]
introduce = "introduceInfo"

# another one-to-one table strategy
[[output.config.routes.one-one]]
match-schema = "test"
match-table = "student_class"
fk-column = "student_id"
include-column = []
exclude-column = []
mode = 1
property-name = "studentClass"
property-pre = "sc_"

[output.config.routes.one-one.convert-column]
name = "className"

# one-to-more table strategy
[[output.config.routes.one-more]]
match-schema = "test"
match-table = "student_parent"
# foreign key column
fk-column = "student_id"
#Included columns, default all
include-column = []
#Excluded column, default empty
exclude-column = []
# Property object name
property-name = "studentParent"

# columns convert strategy
[output.config.routes.one-more.convert-column]
name = "parentName"
```

### CODE REVIEW

**Comment by Ryan-Git** [comment]
Thanks for ur contribution(especially with awesome doc). @ming-relax Let's take a look.

**Comment by Ryan-Git** [comment]
LGTM, @ming-relax any comment?

File: docs/2.0/04-outputs.md [hunk 1]
```diff
@@ -208,4 +208,129 @@ target-type = "_doc" # 默认为 'doc'
...
+[[output.config.routes]]
+match-schema = "test"
+# 主表
+match-table = "student"
+#索引名
+index-name="student_index"
+#类型名，es7该项无效
+type-name="student"
+#分片数
+shards-num=1
```
Comment (ming535) [inline docs/2.0/04-outputs.md:266]: 这里面有和 route 无关的配置吗？能放到 output.config 里面去吗

File: docs/2.0/04-outputs.md [hunk 2]
```diff
@@ -208,4 +208,129 @@ target-type = "_doc" # 默认为 'doc'
...
+replicas-num=0
+#失败重试次数
+retry-count=3
+#包含的列，默认全部
+include-column = []
+#排除的列，默认没有
+exclude-column = []
+
+# 列名转义策略
+[output.config.routes.convert-column]
```
Comment (ming535) [inline docs/2.0/04-outputs.md:277]: convert-column 能做成 filter 吗

**Review by ming535 — CHANGES_REQUESTED** [review]

File: docs/2.0/04-outputs.md (hunk 1 above)
Comment (treeyh) [reply docs/2.0/04-outputs.md:266]: 设计是支持多索引同步，因此配置是放到每个主表索引节点中

File: docs/2.0/04-outputs.md (hunk 2 above)
Comment (treeyh) [reply docs/2.0/04-outputs.md:277]: 考虑到每个索引的转义策略可能不同，因此做成单独配置

File: docs/2.0/04-outputs-en.md [hunk 3]
```diff
@@ -203,4 +203,124 @@ target-type = "_doc" # the default value is 'doc'
...
+exclude-column = []
+mode = 1
+property-name = "studentClass"
+property-pre = ""
+
+[output.config.routes.one-one.convert-column]
+name = "className"
+
+# one-to-more table strategy
+[[output.config.routes.one-more]]
```
Comment (ming535) [inline docs/2.0/04-outputs-en.md]: `one-to-more` --> `one-to-many` ?

Comment (treeyh) [reply docs/2.0/04-outputs-en.md]: 改好了，再看一下

**Review by ming535 — APPROVED** [review]

**Pull request merged**